### PR TITLE
add list of collations that are required to determine equality. 

### PR DESCRIPTION
**Pull request by Tmonster**
For some collations, we cannot do a normal byte comparison for equality. The collation information is needed instead.

"da" Is the only one I know of so far. @carlopi has suggested to write a script to check all collations and character equalities to see if there are other collations that are required to evaluate equality.

### CODE REVIEW

File: src/parser/parsed_data/create_collation_info.cpp [hunk 1]
```diff
@@ -2,11 +2,16 @@
...
 
+static unordered_set<string> const equality_requires_collation = {"da"};
+
 CreateCollationInfo::CreateCollationInfo(string name_p, ScalarFunction function_p, bool combinable_p,
                                          bool not_required_for_equality_p)
     : CreateInfo(CatalogType::COLLATION_ENTRY), function(std::move(function_p)), combinable(combinable_p),
       not_required_for_equality(not_required_for_equality_p) {
 	this->name = std::move(name_p);
+	if (equality_requires_collation.count(this->name) > 0) {
+		this->not_required_for_equality = false;
```
Comment (Mytherin) [inline src/parser/parsed_data/create_collation_info.cpp:13]: This should be fixed where the  `CreateCollationInfo` is created (`IcuExtension::Load` in `icu-extension.cpp`). I would also say let's set this to false for all ICU extensions.

**Comment by guihembichot** [comment]
I hope I can add a comment here. The wrong results with collation "da", are also observed with other ICU collations (tried "de"). It's not due to a special collation.
And, it would be interesting to have, in the test queries which use GROUP BY, the printout of the group column ('s') too, on top of COUNT(*); to verify that the building of groups, which transforms values of 's' into equivalent ones, leaves a final correct 's' after grouping (probably one of the two original strings in the table, but not something else).

File: src/parser/parsed_data/create_collation_info.cpp [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 namespace duckdb {
 
+static unordered_set<string> const equality_requires_collation = {"da"};
```
Comment (Mytherin) [inline src/parser/parsed_data/create_collation_info.cpp:5]: Can we remove this as well?

Comment (Tmonster) [reply src/parser/parsed_data/create_collation_info.cpp]: done

**Review by Mytherin — APPROVED** [review]
Thanks! LGTM

**Pull request merged**